mesonbuild/modules/python.py

```diff
@@ -66,6 +64,9 @@ class FindInstallationKw(ExtractRequired):
 
 
 class PythonExternalProgram(BasicPythonExternalProgram):
+
+    run_bytecompile: T.ClassVar[T.Dict[str, bool]] = {}
```

Comment:
Why does it have to be a dict instead of a direct boolean?

Reply:
The key is the found installation, the value is whether it installed bytecode. This avoids producing an install script for installations that didn't install anything, just because a different installation *did*.

Reply:
I mean, why isn't that just a bool attribute on the installation instance?

```
if not any(i.run_bytecompile for i in self.installations.values()):
  return
```

Reply:
Ah, that's a different question indeed.

I did this the same way that FeatureNew uses a ClassVar cache, to ensure that it's traceable across reinstantiation. I think I was worried about:
- multiple import('python') across subprojects
- multiple find_installations overwriting each other

Let me check, I think both of these are cached nowadays...

Reply:
Sadly it turns out that that was not what I was worried about. I was worried about the underlying PythonExternalProgram object itself. We cache a basic one in _find_installation_impl(), but then need to modify it when returning it, e.g. to set the default `pure:` kwarg.

Clearly this, too, is not well documented if I cannot remember the rationale for it either. I'll add a comment explaining why we need this.

Reply:
oh I see `python = copy.copy(python)`, a comment would do then indeed.

Reply:
Note it could be a `T.Set[str]` instead of dict mapping to bool.

Reply:
I just pushed a comment, anyway. :)